### DISCUSSION TITLE
레벨3 회고

### DISCUSSION THREAD

**Discussion by zeus6768** [woowacourse-teams/2024-code-zap#574] [이야기해봐요]
1. 방학 때 개발 할 건지
2. 배포 및 마케팅
3. 레벨4 참여비중
4. 회고

**Comment by zeus6768** [comment]
# 안건 및 결정

## 안건1. : 방학에 개발 할 건지?
- 결정: 방학에는 프로젝트 작업하지 않아도 된다. 

## 안건2: 배포 및 마케팅
- 결정
  - 방학에는 배포하지 않는다. 
  - 레벨4 첫주 9월 6일 금요일 배포
    - 불특정 다수 사용자에게 배포

## 안건3: 참여비중 - 데일리 스크럼 할 건지?
- 결정
  - 9월 3~6일 매일 아침 데일리 미팅 유지
  - 9월 9일에 코어타임 회의 예정

## 안건4: 회고

### 초롱
- QA를 더 자주 해야 할 것 같다. 
- 기능을 추가하는 대신에 리팩토링에 리소스를 사용했다면 테스트가 보완됐을 것 같다. 

### 짱수
- 개발 담당자가 QA를 해야 한다. 

### 제우스
- 이슈에 기능 목록을 체크리스트로 만들어서 하는 것처럼, QA를 체크리스트에 따라 하자. 
- QA기간에는 QA만 해야 한다. 

### 마위
- E2E 테스트를 해야 한다. 

### 켬미
- PR 단위가 너무 컸기 때문에 코드리뷰의 퀄리티가 떨어졌다. 
- 그 과정에서 누락되거나 잘못된 로직이 머지됐다. 
- 테스트 리팩토링을 했는데도 버그가 발견되지 않았다. 왜 그랬을까?
  - 기능 명세가 상세하지 않았기 때문이다. 

### 몰리
- 테스트가 더 꼼꼼히 작성되어야 한다. 
- 레벨1, 2에는 요구사항이 주어지면 요구사항을 구체적으로 작성했다. 레벨3에는 그렇게 하지 않아서 그렇다. 

### 헤인
- 이슈에 있는 코멘트 기능을 활용하자.

## 안건5: 블로그
- 결정
  - 헤인이 블로그 하나 만들어서 가져옴

## 안건6: 코드리뷰 규칙
### 24시간 또는 36시간 내 리뷰
- 결정
  - 36시간 내 코드리뷰
### PR 마감 시간 규칙
- 결정
  - ZAP 스프린트를 지키자. 

## 안건7: 모든 팀원이 프로젝트 개발 아키텍처를 발표하는 시간을 갖자
- 발제자: 마위
- 결정:
  - 9월 9일에 회의해서 세부 내용을 정한다. 

## 안건8: 사용하지 않는 브랜치 모두 삭제 및 이슈 close

**Reply by zeus6768** [reply]
### 안건 8 관련 브랜치 목록
2024.09.03 11:20 기준 브랜치
형식: {브랜치명} {마지막 커밋 작성자}
- `refactor/521-error-code` 켬미
- `refactor/476-test-division-service` 짱수
- `refactor/logback_dev` 초롱
- `refactor/fe-cd-separate-beta` 월하
- `fix/510-v1.1.0-conflict` 마스터위
- `fix/510-type-error-on-category-upload` 월하
- `fix/revert_dev/fe` 마스터위
- `fix/frontend_cd_rm` 마스터위
- `fix/cd_front` 초롱
- `feat/510-login-feedback` 마스터위
- `feat/495-find-templates` 짱수
- `feat/442-selectList-io` 헤인
- `feat/template_explore` 월하
- `feat/category_edit` 월하

\+ develop 브랜치 삭제 제안

**Comment by zangsu** [comment]
제이슨이 공유해 준 이상적인 이슈 예시입니다!

- https://github.com/woowacourse/service-apply/issues/565
- https://github.com/woowacourse/service-apply/issues/738
